### PR TITLE
Add support for POST from the Firebase REST API

### PR DESCRIPTION
**Pull request by davenotik**
Support for POST, and associated tests. Also, test formatting cleaned up for readability (group related lines of code together) and dartfmt applied.

I presume you'll want me to squash these commits to one. Perhaps you can recommend the best method  as I've already pushed to my forked repo?

Later, support for PATCH. As well, `decodeKey` and `encodeKey` aren't quite working and I so I used the version I'd previously written. I'll look at those soon.

### CODE REVIEW

File: test/firebase_io_test.dart [hunk 1]
```diff
@@ -11,66 +13,63 @@ Uri getTestUrl(int count, {DateTime timeKey}) => getTestUrlBase(
...
       expect(rootValue, putContent);
+
+      // TODO: Consider improving how we construct this new path.
+      var postUri = Uri
+          .parse('${baseUri.toString().split('.json').first}/interests.json');
+      var postContent = [
+        {'name': 'Encryption', 'love-level': 10},
+        {'name': 'Computer Science', 'love-level': 7}
+      ];
+      await Future.forEach(postContent, (interest) async {
```
Comment (kevmoo) [inline test/firebase_io_test.dart]: Since you're in an async method, just use a normal for loop

File: test/firebase_io_test.dart [hunk 2]
```diff
@@ -11,66 +13,63 @@ Uri getTestUrl(int count, {DateTime timeKey}) => getTestUrlBase(
 //TODO: add tests that validate security by having secured and insecure sections
 void main() {
   int count = 0;
-  group('get and put', () {
+  group('get, put and post', () {
     test('without security', () async {
       var baseUri = getTestUrl(count++);
-
```
Comment (kevmoo) [inline test/firebase_io_test.dart]: undo all of this whitespace cleanup – ideally your commit touches minimal other code

**Comment by kevmoo** [comment]
@davenotik Great to see this!

Yes, please squash your pull request

* Do `git rebase -i origin/master` on your local machine so you have one commit
* Do a `git push --force davenotik` or whatever you named your folk

**Comment by kevmoo** [comment]
This is a good reason why it's good to work on a branch – and push your working branch – so you don't accidentally mess up master

**Comment by davenotik** [comment]
I guess it opened a new PR as I switched to a new branch. See:
https://github.com/firebase/firebase-dart/pull/68

**Pull request closed without merging**